doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,52 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+######
+
+.. _COSIM:VHPIDIRECT:Examples:arrays:matrices:
+
+:cosimtree:`matrices <vhpidirect/arrays/matrices>`
+**************************************************
+
+Constrained multidimensional arrays of doubles/reals
+====================================================
+
+As explained in :ref:`Restrictions_on_foreign_declarations`, unconstrained arrays are represented by a fat pointer in C,
+and it is not suggested to use fat pointers unless it is unavoidable. Hence, it is desirable for matrices which are to
+be represented as multidimensional arrays to be constrained. However, constraining multidimensional arrays in VHDL and
+C separatedly is error prone.
+
+This example shows a general procedure to share constrained multidimensional arrays of any size. Dimensions of a 2D
+matrix of doubles are defined in C and a helper function is used for VHDL to read those values into the declaration of
+an *array of reals* type. Then, the pointer to the matrix (in C) is retrieved as an access (in VHDL), through another
+helper function. For completeness, IEEE's ``fixed_generic_pkg`` package is used to multiply each value with a constant
+using fixed-point formats.
+
+.. HINT::
+  Different types need to be declared in VHDL, to use this approach for passing multiple matrices of differentes
```

Comment:
"multiple matrices of different sizes"

suggestion: A VHDL access that is used for a bounded array is tied to a type that specifies its length (in this case 2 dimensional lengths are specified). This means that a separate type would need to be declared for an array with a different size, different dimensions or different sizes in any dimension. In order to create a single type for many differently shaped arrays, it would...

I think that that comment could be moved up the line of examples  and referenced a number of times.

Reply:
I updated it according to your suggestion and I moved it to the top of page/section. Also, I changed it to IMPORTANT.